### DISCUSSION TITLE
Update firmware.

### DISCUSSION THREAD

**Discussion by Wallat1** [bitcraze/discussions#352] [Q&A] [answered]
Hi.

I have seen than to update the firmware of the crazyflie it is necessary to enter Makefile and in the following lines 

 ######### Stabilizer configuration ##########
ESTIMATOR          ?= any
CONTROLLER         ?= Any # one of Any, PID, Mellinger, INDI
POWER_DISTRIBUTION ?= stock
 
it is required to replace "Any" for one of the three controllers but now the previous lines do not appear in the file Makefile, for this reason I would like to know what is the new method  to update the controllers?.

Regards.

**Comment by knmcguire** [comment]
Hi! No this needs to be configured with kbuild now and 'make menuconfig'. Please check this documentation: https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/development/kbuild/

**Reply by knmcguire** [reply]
Did this solve your problem? if it did, please mark it as an answer or give us an update

**Reply by Wallat1** [reply]
Hi.

Sorry for not responding earlier, but I haven't had a chance to try what's indicated in the link yet, I'll try it tomorrow and if it works I'll let you know.

Regards.

**Comment by Wallat1** [comment]
Hi.

Sorry for not answering, but until today I can make use of the CF. Regarding my problem, I have already done what is indicated in the documentation, however, the .config file is created inside the build folder, for this reason I would like to know if that is ok? and if this is correct, what would be the next step to load the firmware to the CF?.

Regards.

**Reply by knmcguire** [reply]
That probably seems about right. If you build the firmware and flash with  'make cload' (after putting the crazyflie in bootloader mode), then you could check in the parameter tab of the cfclient if the right controller has been selected by standard.

**Reply by Wallat1** [reply]
Hi.
I have followed your instructions, but in the Ubuntu terminal a message appears that says .config file not found, I have cloned the repository https://github.com/bitcraze/crazyflie-firmware
again and modified the Kconfig file but the problem persists, could you please tell me if it is necessary to execute any other command in the terminal?.

Regards.

**Comment by knmcguire** [comment, marked as answer]
Ah no you shouldn't change the Kconfig file itself. Please revert that change to the original state and only configure things through the instructions linked to above.

So here is the full step:

1. `make clean`
2. `make cf2_defconfig` (probably not necessary in your case but just to be sure)
3. `make menuconfig`
4. Once you see the menu appear, go to 'controllers and estimators' >'Default controller' > now select which controller you would like to select and hit spacebar. 
5. use the arrow keys to go to 'exit' , hit enter two times until you see 'do you wish to save your new configuration?', so in your case it is yes.
6. Then do `make -j 12`
7. Then flash the firmware again with `make cload`

**Reply by Wallat1** [reply]
Hi.

Thank you so much for your help,  I have already been able to load the firmware.

Regards.
